### PR TITLE
Sanity check for surveyed position.

### PR DESCRIPTION
**Pull request by ungureanuvladvictor**
I'm not really sure what the distance threshold should be. 

/cc: @denniszollo @fnoble 
Signed-off-by: Vlad Ungureanu <vvu@vdev.ro>

### CODE REVIEW

File: src/system_monitor.c [hunk 1]
```diff
@@ -28,10 +30,13 @@
 #include "manage.h"
 #include "simulator.h"
 #include "system_monitor.h"
+#include "position.h"
 
 #define WATCHDOG_HARDWARE_PERIOD_MS 30000  /* Actual period may vary +88% -32% */
 #define WATCHDOG_THREAD_PERIOD_MS 15000
 
+#define BASE_STATION_DISTANCE_THRESHOLD 15000
```
Comment (fnoble) [inline src/system_monitor.c]: Comment with units?

File: src/system_monitor.c [hunk 2]
```diff
@@ -161,8 +166,23 @@ static msg_t system_monitor_thread(void *arg)
...
+
+      llhdeg2rad(base_llh, tmp);
+      wgsllh2ecef(tmp, base_ecef);
+
+      vector_subtract(3, base_ecef, position_solution.pos_ecef, tmp);
+      base_distance = vector_norm(3, tmp);
+
+      if (base_distance > BASE_STATION_DISTANCE_THRESHOLD) {
+        log_warn("Invalid surveyed position coordinates!\n");
+      }
```
Comment (fnoble) [inline src/system_monitor.c]: Linux Kernely style says use `} else {`

File: src/system_monitor.c [hunk 3]
```diff
@@ -161,8 +167,22 @@ static msg_t system_monitor_thread(void *arg)
...
+      double base_distance;
+
+      llhdeg2rad(base_llh, tmp);
+      wgsllh2ecef(tmp, base_ecef);
+
+      vector_subtract(3, base_ecef, position_solution.pos_ecef, tmp);
+      base_distance = vector_norm(3, tmp);
+
+      if (base_distance > BASE_STATION_DISTANCE_THRESHOLD) {
+        log_warn("Invalid surveyed position coordinates!\n");
```
Comment (denniszollo) [inline src/system_monitor.c]: Nit- no need for ! character.

File: src/system_monitor.c [hunk 4]
```diff
@@ -28,10 +30,14 @@
 #include "manage.h"
 #include "simulator.h"
 #include "system_monitor.h"
+#include "position.h"
 
 #define WATCHDOG_HARDWARE_PERIOD_MS 30000  /* Actual period may vary +88% -32% */
 #define WATCHDOG_THREAD_PERIOD_MS 15000
 
+/* Distance in meters between rover and surveyed base station position */
```
Comment (cbeighley) [inline src/system_monitor.c]: Isn't this the distance between the surveyed and calculated base station position? In which case something like
```
/* Maximum distance between calculated and surveyed base station single point position for error checking. */
```

**Comment by cbeighley** [comment]
lgtm

File: src/system_monitor.c (hunk 1 above)
Comment (denniszollo) [inline src/system_monitor.c]: I assume these are meters?
so the position must be within 15 Kilomters?

Comment (ungureanuvladvictor) [inline src/system_monitor.c]: Yes

**Pull request merged**